src/plugins/data/public/query/query_string/query_string_manager.ts

```diff
@@ -279,7 +286,11 @@ export class QueryStringManager {
       ...curQuery,
       language: languageId,
     };
-    const queryString = language?.getQueryString(newQuery) || '';
+    let queryString = language?.getQueryString(newQuery) || '';
```

Comment:
isn't this duplicating what getInitialDatasetQueryString is doing?

Reply:
I can put this behind else

Reply:
Yeah, `queryString = curQuery.dataset ? this.getInitialDatasetQueryString(newQuery, curQuery.dataset) : (language?.getQueryString(newQuery) || '')` might make this a little less confusing. Or just handling the undefined dataset case in the common method (preferred imo, but not sure I'm missing something)